src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -3064,6 +3064,347 @@ public static void Main()
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void AttributesOnPrimaryConstructor_09([CombinatorialValues("class", "struct", "record", "record class", "record struct")] string declaration)
+        {
+            string source = @"
+[System.AttributeUsage(System.AttributeTargets.All, AllowMultiple = true) ]
+public class A : System.Attribute
+{
+}
+
+[System.AttributeUsage(System.AttributeTargets.All, AllowMultiple = true) ]
+public class B : System.Attribute
+{
+}
+
+[method: A]
+partial " + declaration + @" C1();
+
+#line 100
+[method: B]
+partial " + declaration + @" C1
+#line 200
+    ();
+
+[method: B]
+partial " + declaration + @" C2();
+
+#line 300
+[method: A]
+partial " + declaration + @" C2
+#line 400
+    ();
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (100,2): warning CS0657: 'method' is not a valid attribute location for this declaration. Valid attribute locations for this declaration are 'type'. All attributes in this block will be ignored.
+                // [method: B]
+                Diagnostic(ErrorCode.WRN_AttributeLocationOnBadDeclaration, "method").WithArguments("method", "type").WithLocation(100, 2),
+                // (200,5): error CS8863: Only a single partial type declaration may have a parameter list
+                //     ();
+                Diagnostic(ErrorCode.ERR_MultipleRecordParameterLists, "()").WithLocation(200, 5),
+                // (300,2): warning CS0657: 'method' is not a valid attribute location for this declaration. Valid attribute locations for this declaration are 'type'. All attributes in this block will be ignored.
+                // [method: A]
+                Diagnostic(ErrorCode.WRN_AttributeLocationOnBadDeclaration, "method").WithArguments("method", "type").WithLocation(300, 2),
+                // (400,5): error CS8863: Only a single partial type declaration may have a parameter list
+                //     ();
+                Diagnostic(ErrorCode.ERR_MultipleRecordParameterLists, "()").WithLocation(400, 5)
+                );
+
+            var c1 = (SourceNamedTypeSymbol)comp.GetTypeByMetadataName("C1");
+            Assert.Empty(c1.GetAttributes());
+            Assert.True(c1.HasPrimaryConstructor);
+            Assert.Equal("A", c1.PrimaryConstructor.GetAttributes().Single().ToString());
+            Assert.True(c1.Constructors.Where(ctor => ctor != c1.PrimaryConstructor).All(ctor => ctor.GetAttributes().IsEmpty));
+
+            var c2 = (SourceNamedTypeSymbol)comp.GetTypeByMetadataName("C2");
+            Assert.Empty(c2.GetAttributes());
+            Assert.True(c2.HasPrimaryConstructor);
+            Assert.Equal("B", c2.PrimaryConstructor.GetAttributes().Single().ToString());
+            Assert.True(c2.Constructors.Where(ctor => ctor != c2.PrimaryConstructor).All(ctor => ctor.GetAttributes().IsEmpty));
+        }
```

Comment:
I assume it's covered already (and orthogonal to the attribute target) but consider also testing some attributes with `nameof(parameter)`